bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/handler/OpenThermGatewayHandler.java

```diff
@@ -0,0 +1,305 @@
+
+    @Override
+    public void log(LogLevel loglevel, String message) {
+        switch (loglevel) {
+            case Trace:
+                logger.trace(message);
+                break;
+            case Debug:
+                logger.debug(message);
+                break;
+            case Info:
+                logger.info(message);
+                break;
+            case Warning:
+                logger.warn(message);
+                break;
+            case Error:
+                logger.error(message);
+                break;
+            default:
+                break;
+        }
+    }
+
+    @Override
+    public void handleRemoval() {
+        logger.debug("Removing OpenTherm Gateway handler");
+        disconnect();
+        super.handleRemoval();
+    }
+
+    @Override
+    public void dispose() {
+        disconnect();
+        super.dispose();
+    }
+
+    private boolean checkConnection() {
+        if (connector != null && connector.isConnected()) {
+            return true;
+        }
+
+        return connect();
+    }
+
+    private synchronized boolean connect() {
+        try {
+            disconnect();
+
+            logger.info("Starting OpenTherm Gateway connector");
+
+            explicitDisconnect = false;
+            // TODO: support different kinds of connectors, such as USB, serial port
+            connector = new OpenThermGatewaySocketConnector(this, config.ipaddress, config.port);
+            new Thread(connector).start();
+
+            logger.info("OpenTherm Gateway connector started");
+
+            return true;
+        } catch (Exception ex) {
+            logger.error("error", ex);
+        }
+
+        return false;
+    }
+
+    boolean explicitDisconnect = false;
+
+    private synchronized void disconnect() {
+        if (connector != null) {
+            if (connector.isConnected()) {
+                logger.info("Stopping OpenTherm Gateway connector");
+
+                explicitDisconnect = true;
+                connector.stop();
+            }
+
+            connector = null;
+        }
+    }
+
+    private String getGatewayCodeFromChannel(String channel) throws Exception {
+        switch (channel) {
+            case OpenThermGatewayBindingConstants.CHANNEL_OVERRIDE_SETPOINT_TEMPORARY:
+                return GatewayCommandCode.TemperatureTemporary;
+            case OpenThermGatewayBindingConstants.CHANNEL_OVERRIDE_SETPOINT_CONSTANT:
+                return GatewayCommandCode.TemperatureConstant;
+            case OpenThermGatewayBindingConstants.CHANNEL_OUTSIDE_TEMPERATURE:
+                return GatewayCommandCode.TemperatureOutside;
+            case OpenThermGatewayBindingConstants.CHANNEL_OVERRIDE_DHW_SETPOINT:
+                return GatewayCommandCode.SetpointWater;
+            case OpenThermGatewayBindingConstants.CHANNEL_SEND_COMMAND:
+                return "";
+            default:
+                throw new Exception(String.format("Unknown channel %s", channel));
```

Comment:
Please don't throw Exception. Either use `IllegalArgumentException` (in this case) or inherit a new exception class. Also applies to ther classes)